### PR TITLE
Further refactor person to student

### PR DESCRIPTION
**Pull request by fatin99**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -41,65 +41,64 @@
...
             + PREFIX_PHONE + "91234567 "
-            + PREFIX_EMAIL + "johndoe@example.com"
+            + PREFIX_EMAIL + "johndoe@example.com "
             + PREFIX_MATRIC + "A0181234J";
 
-    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Student: %1$s";
+    public static final String MESSAGE_EDIT_STUDENT_SUCCESS = "Edited Student: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This student already exists in the address book.";
+    public static final String MESSAGE_STUDENT_STUDENT = "This student already exists in the address book.";
```
Comment (fatin99) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: MESSAGE_DUPLICATE_STUDENT

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 2]
```diff
@@ -15,7 +15,6 @@
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.EditCommand;
-import seedu.address.logic.commands.EditCommand.EditPersonDescriptor;
```
Comment (fatin99) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:18]: Missing import, not replaced

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 3]
```diff
@@ -42,26 +41,26 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
+        EditCommand.EditStudentDescriptor editStudentDescriptor = new EditCommand.EditStudentDescriptor();
```
Comment (fatin99) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java]: User editStudentDescriptor, not editCommand

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 4]
```diff
@@ -83,75 +83,75 @@ public void execute_noFieldSpecifiedUnfilteredList_success() {
...
-        Student firstStudent = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
-        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder(firstStudent).build();
-        EditCommand editCommand = new EditCommand(INDEX_SECOND_PERSON, descriptor);
+    public void execute_duplicateStudentUnfilteredList_failure() {
+        Student firstStudent = model.getFilteredStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
+        EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder(firstStudent).build();
+        EditCommand editCommand = new EditCommand(INDEX_SECOND_STUDENT, descriptor);
 
-        assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_PERSON);
+        assertCommandFailure(editCommand, model, EditCommand.MESSAGE_STUDENT_STUDENT);
```
Comment (fatin99) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java]: Wrong message name

File: src/test/java/seedu/address/logic/commands/EditStudentDescriptorTest.java [hunk 5]
```diff
@@ -36,23 +36,24 @@ public void equals() {
         assertFalse(DESC_AMY.equals(DESC_BOB));
 
         // different name -> returns false
-        EditPersonDescriptor editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withName(VALID_NAME_BOB).build();
+        EditCommand.EditStudentDescriptor editedAmy = new EditStudentDescriptorBuilder(DESC_AMY)
```
Comment (fatin99) [inline src/test/java/seedu/address/logic/commands/EditStudentDescriptorTest.java]: remove reference to editcommand

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 6]
```diff
@@ -135,46 +135,47 @@ public void parse_someFieldsSpecified_success() {
     @Test
     public void parse_oneFieldSpecified_success() {
         // name
-        Index targetIndex = INDEX_THIRD_PERSON;
+        Index targetIndex = INDEX_THIRD_STUDENT;
         String userInput = targetIndex.getOneBased() + NAME_DESC_AMY;
-        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withName(VALID_NAME_AMY).build();
+        EditCommand.EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder()
```
Comment (fatin99) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: Shouldnt reference EditCommand when using EditStudentDescriptor

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 7]
```diff
@@ -135,46 +135,47 @@ public void parse_someFieldsSpecified_success() {
...
     public void parse_multipleRepeatedFields_acceptsLast() {
-        Index targetIndex = INDEX_FIRST_PERSON;
+        Index targetIndex = INDEX_FIRST_STUDENT;
         String userInput = targetIndex.getOneBased() + PHONE_DESC_AMY + EMAIL_DESC_AMY
                 + MATRIC_DESC_AMY + TAG_DESC_FRIEND + PHONE_DESC_AMY + EMAIL_DESC_AMY
                 + MATRIC_DESC_AMY + TAG_DESC_FRIEND + PHONE_DESC_BOB + EMAIL_DESC_BOB
                 + MATRIC_DESC_BOB + TAG_DESC_HUSBAND;
 
-        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withPhone(VALID_PHONE_BOB)
+        EditCommand.EditStudentDescriptor descriptor = new EditStudentDescriptorBuilder().withPhone(VALID_PHONE_BOB)
```
Comment (fatin99) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: Shouldnt reference EditCommand when using EditStudentDescriptor

File: src/test/java/seedu/address/testutil/EditStudentDescriptorBuilder.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+    public EditStudentDescriptorBuilder(EditCommand.EditStudentDescriptor descriptor) {
+        this.descriptor = new EditCommand.EditStudentDescriptor(descriptor);
+    }
+
+    /**
+     * Returns an {@code EditStudentDescriptor} with fields containing {@code student}'s details
+     */
+    public EditStudentDescriptorBuilder(Student student) {
+        descriptor = new EditCommand.EditStudentDescriptor();
```
Comment (fatin99) [inline src/test/java/seedu/address/testutil/EditStudentDescriptorBuilder.java]: should not reference editcommand

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java (hunk 2 above)
Comment (PotatoCombat) [reply src/main/java/seedu/address/logic/parser/EditCommandParser.java:18]: EditPersonDescriptor -> EditStudentDescriptor. Okie!

File: src/main/java/seedu/address/model/student/Student.java [hunk 9]
```diff
@@ -61,10 +61,10 @@ public Matric getMatric() {
...
-     * Returns true if both persons of the same name have at least one other identity field that is the same.
-     * This defines a weaker notion of equality between two persons.
+     * Returns true if both students of the same name have at least one other identity field that is the same.
+     * This defines a weaker notion of equality between two students.
      */
-    public boolean isSamePerson(Student otherStudent) {
+    public boolean isSameStudent(Student otherStudent) {
         if (otherStudent == this) {
             return true;
         }
```
Comment (PotatoCombat) [inline src/main/java/seedu/address/model/student/Student.java:70]: Remember to update this stub later on!

File: src/main/resources/view/DarkTheme.css [hunk 10]
```diff
@@ -328,7 +328,7 @@
     -fx-text-fill: white;
 }
 
-#filterField, #personListPanel, #personWebpage {
+#filterField, #studentListPanel, #studentWebpage {
```
Comment (PotatoCombat) [inline src/main/resources/view/DarkTheme.css:331]: Will need to double check what Webpage here does.

**Review by PotatoCombat — APPROVED** [review]

**Pull request merged**